### PR TITLE
setup CI

### PR DESCRIPTION
**Pull request by manimotion**
- Creating the main.yml
- Editing the Cypres.config.ts to turn off the record of videos.

### CODE REVIEW

File: cypress.config.ts [hunk 1]
```diff
@@ -2,8 +2,16 @@ const { defineConfig } = require("cypress");
...
+    chromeWebSecurity: false,
     setupNodeEvents(on, config) {
-      // implement node event listeners here
-    },
+    // e2e testing node events setup code
+    config.defaultCommandTimeout= 10000;
+    config.responseTimeout= 20000;
+    config.pageLoadTimeout= 60000;
+    return config;
   },
```
Comment (holgiosalos) [inline cypress.config.ts]: fix the indentation for these lines, as they are part of the setupNodeEvents, you should add one indent more.

File: package.json [hunk 2]
```diff
@@ -4,9 +4,11 @@
...
   "main": "index.js",
   "scripts": {
-    "test": "cypress open",
-    "test:open": "cypress open"
-  },
+    "test:open": "npm run lint && cypress open",
+    "test": "cypress run --headless --browser chrome",
+    "lint": "eslint ./cypress/e2e/**/*.ts",
+    "lint:fix": "npm run lint -- --fix"
+},
```
Comment (holgiosalos) [inline package.json]: fix the indentation for line 11

Comment (manimotion) [reply package.json]: Solved

File: cypress.config.ts (hunk 1 above)
Comment (manimotion) [reply cypress.config.ts]: Solved

**Review by holgiosalos — APPROVED** [review]

**Pull request merged**